pkg/cmd/server/kubernetes/master/master_config.go

```diff
@@ -456,52 +456,12 @@ func buildKubeApiserverConfig(
 
 	// we don't use legacy audit anymore
 	genericConfig.LegacyAuditWriter = nil
-	if masterConfig.AuditConfig.Enabled {
-		var writer io.Writer
-		if len(masterConfig.AuditConfig.AuditFilePath) > 0 {
-			writer = &lumberjack.Logger{
-				Filename:   masterConfig.AuditConfig.AuditFilePath,
-				MaxAge:     masterConfig.AuditConfig.MaximumFileRetentionDays,
-				MaxBackups: masterConfig.AuditConfig.MaximumRetainedFiles,
-				MaxSize:    masterConfig.AuditConfig.MaximumFileSizeMegabytes,
-			}
-		} else {
-			// backwards compatible writer to regular log
-			writer = cmdutil.NewGLogWriterV(0)
-		}
-		genericConfig.AuditBackend = auditlog.NewBackend(writer, auditlog.FormatLegacy)
-		genericConfig.AuditPolicyChecker = auditpolicy.NewChecker(&auditinternal.Policy{
-			// This is for backwards compatibility maintaining the old visibility, ie. just
-			// raw overview of the requests comming in.
-			Rules: []auditinternal.PolicyRule{{Level: auditinternal.LevelMetadata}},
-		})
-
-		// when a policy file is defined we enable the advanced auditing
-		if masterConfig.AuditConfig.PolicyConfiguration != nil || len(masterConfig.AuditConfig.PolicyFile) > 0 {
-			// policy configuration
-			if masterConfig.AuditConfig.PolicyConfiguration == nil {
-				p, _ := auditpolicy.LoadPolicyFromFile(masterConfig.AuditConfig.PolicyFile)
-				genericConfig.AuditPolicyChecker = auditpolicy.NewChecker(p)
-			} else if len(masterConfig.AuditConfig.PolicyFile) > 0 {
-				p := masterConfig.AuditConfig.PolicyConfiguration.(*auditinternal.Policy)
-				genericConfig.AuditPolicyChecker = auditpolicy.NewChecker(p)
-			}
-
-			// log configuration, only when file path was provided
-			if len(masterConfig.AuditConfig.AuditFilePath) > 0 {
-				genericConfig.AuditBackend = auditlog.NewBackend(writer, string(masterConfig.AuditConfig.LogFormat))
-			}
-
-			// webhook configuration, only when config file was provided
-			if len(masterConfig.AuditConfig.WebHookKubeConfig) > 0 {
-				webhook, err := auditwebhook.NewBackend(masterConfig.AuditConfig.WebHookKubeConfig, string(masterConfig.AuditConfig.WebHookMode))
-				if err != nil {
-					glog.Fatalf("Audit webhook initialization failed: %v", err)
-				}
-				genericConfig.AuditBackend = audit.Union(genericConfig.AuditBackend, webhook)
-			}
-		}
+	backend, policyChecker, err := GetAuditConfig(masterConfig.AuditConfig)
+	if err != nil {
+		glog.Fatalf("Audit initialization failed: %v", err)
```

Comment:
method returns an error

Reply:
@soltysh ?

Reply:
I swear I fixed that. Must have been lost when going back and forth on this piece. I've opened https://github.com/openshift/origin/pull/16594 to address that.